src/model/entities/rundown.ts

```diff
@@ -178,15 +189,17 @@ export class Rundown extends BasicRundown {
     this.nextCursor.part.setAsNext()
   }
 
-  private findNextSegment(): Segment {
+  private findNextValidSegment(): Segment {
     const activeSegmentIndex: number = this.segments.findIndex((segment) => segment.id === this.activeCursor?.segment?.id)
     if (activeSegmentIndex === -1) {
-      throw new NotFoundException('Segment does not exist in Rundown')
+      throw new NotFoundException('Active Segment does not exist in Rundown')
     }
-    if (activeSegmentIndex === this.segments.length + 1) {
-      throw new LastSegmentInRundown(`Segment: ${this.activeCursor?.segment?.id} is the last Segment of Rundown: ${this.id}`)
+    for (let i = activeSegmentIndex; i < this.segments.length - 1; i++) {
+      if (this.isSegmentValidForRundownExecution(this.segments[i + 1])) {
+        return this.segments[i + 1]
+      }
     }
-    return this.segments[activeSegmentIndex + 1]
+    throw new LastSegmentInRundown(`Segment: ${this.activeCursor?.segment?.id} is the last Segment of Rundown: ${this.id}`)
```

Comment:
In the exception message, it is stated that it is the last segment in the rundown, but it is the last _valid_ segment in the rundown – minor thing :) Personally, I would like our error messages to be somewhat written out like our construct naming conventions, to ensure a lower level of ambiguity.

The following is just a suggestion. One could use `Array.slice` and `Array.find` instead of the for-loop to find the next valid segment.

```typescript
const nextValidSegment: Segment | undefined = this.segments.slice(activeSegmentIndex + 1).find(segment => this.isSegmentValidForRundownExecution(segment))
if (!nextValidSegment) {
  throw new LastSegmentInRundown(`The segment '${this.activeCursor?.segment.name}' with the id '${this.activeCursor?.segment.id}' is the last valid segment in the Rundown '${this.name}' with the id '${this.id}'.`)
}
return nextValidSegment
```

Reply:
Updated to the `Array.slice` and `Array.find` suggestion.

Changed the error message to include "valid Segment".